### DISCUSSION TITLE
Branch disappeared

### DISCUSSION THREAD

**Discussion by lastanna** [community/community#125154] [Code Security]
### Select Topic Area

Bug

### Body

Hello there,
One of my branches disappear from my project. Changes are not shown in Activities. Perhaps some hackers attack. How could I restore my changes?

**Comment by Janethq** [comment]
If you have accidentally deleted a local Git branch and haven't pushed it to a remote server yet, you can recover it using Git's reflog feature in 2 steps.

1. First, you need to find the commit hash of the last commit in the deleted branch. You can do this by running the following command:

git reflog

This command will display a list of all the commits that have been created or modified in your local repository. Look for the commit hash that corresponds to the last commit in the deleted branch.

2. Once you have the commit hash, you can create a new branch from it using the following command:

git checkout -b <new-branch-name> <commit-hash>

Replace <new-branch-name> with the name you want to give to the new branch, and <commit-hash> with the actual commit hash you found in step 1.

Now you have a new branch that contains all the changes that were in the deleted branch. You can push this branch to the remote repository to make sure you don't lose the changes again.
